application/utils/harvester/change_detector.py

```
@@ -0,0 +1,107 @@
+from .git_repository_client import GitRepositoryClient
+
+logger = logging.getLogger(__name__)
+
+
+class ChangeDetector:
+    def __init__(self, repository_client: GitRepositoryClient):
+        self.repository_client = repository_client
+
+    def _resolve_commit(self, commit_sha: str) -> str:
+        try:
+            result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)